lib/task_helpers/imports/tags.rb

```diff
@@ -0,0 +1,93 @@
+module TaskHelpers
+  class Imports
+    class Tags
+      class ClassificationDescError < StandardError; end
+      class ClassificationNameError < StandardError; end
+      class ClassificationEntryDescError < StandardError; end
+      class ClassificationEntryNameError < StandardError; end
+      class ClassificationYamlError < StandardError
+        attr_accessor :details
+
+        def initialize(message = nil, details = nil)
+          super(message)
+          self.details = details
+        end
+      end
+
+      def import(options = {})
+        return unless options[:source]
+
+        glob = File.file?(options[:source]) ? options[:source] : "#{options[:source]}/*.yaml"
+        Dir.glob(glob) do |fname|
+          begin
+            tag_categories = YAML.load_file(fname)
+            import_tags(tag_categories)
+          rescue ClassificationDescError
+            warn("Error importing #{fname} : Tag category description is required")
+          rescue ClassificationNameError
+            warn("Error importing #{fname} : Tag category name is required")
+          rescue ClassificationEntryDescError
+            warn("Error importing #{fname} : Tag entry description is required")
+          rescue ClassificationEntryNameError
+            warn("Error importing #{fname} : Tag entry name is required")
+          rescue ClassificationYamlError => e
+            warn("Error importing #{fname} : #{e.message}")
+            e.details.each { |k, v| warn("#{k}: #{v.first[:error]}") }
+          rescue ActiveModel::UnknownAttributeError => e
+            warn("Error importing #{fname} : #{e.message}")
+          end
+        end
+      end
+
+      private
+
+      # Description attribute of Tag Categories that are not visible in the UI
+      SPECIAL_TAGS = ['Parent Folder Path (VMs & Templates)', 'Parent Folder Path (Hosts & Clusters)', 'User roles'].freeze
```

Comment:
Again, hardcoded strings that we might accidentally change elsewhere that might cause problems in this script.  I can't tell.

Reply:
See comment on the same issue for the export script above. The same applies.

Reply:
See suggestion above ...